spec/booking_system_spec.rb

```diff
@@ -0,0 +1,328 @@
+      expect(hotel_booker).must_be_instance_of Hotel::BookingSystem
+    end
+
+    it 'if reservations passed in, it stores all reservations in an array of Reservations' do
+      hotel_booker.all_reservations.each do |reservation|
+        expect(reservation).must_be_instance_of Hotel::Reservation
+      end
+    end
+  end
+
+  describe 'list_rooms method' do
+    it 'returns a formatted string of all the rooms' do
+      expect(hotel_booker.list_rooms).must_be_instance_of String
+    end
+  end
+
+  describe 'list_printable_available_rooms method' do
+    it 'returns a formatted string of all the available rooms' do
+      available_rooms = hotel_booker.list_printable_available_rooms(today, today + 7)
+      expect(available_rooms).must_be_instance_of String
+    end
+  end
+
+  describe 'list_available_rooms helper method' do
+
+    it 'returns an array containing a list of available room numbers for a given date range' do
+      expect(hotel_booker.list_available_rooms(captures_both_res_dates)).must_be_instance_of Array
+    end
+
+    it 'correctly identifies the list of available room numbers for a given date range' do
+      all_rooms = [*1..20]
+      all_but_room1 = [*2..20]
+      all_but_room2 = [1, *3..20]
+      all_other_rooms = [*3..20]
+
+      expect(hotel_booker.list_available_rooms(captures_both_res_dates)).must_equal all_other_rooms
+      expect(hotel_booker.list_available_rooms(overlap_res_1)).must_equal all_but_room1
+      expect(hotel_booker.list_available_rooms(overlap_res_2)).must_equal all_but_room2
+
+      expect(hotel_booker.list_available_rooms(before_dates)).must_equal all_rooms
+      expect(hotel_booker.list_available_rooms(after_dates)).must_equal all_rooms
+    end
+
+    it 'correctly identifies the list of available room numbers for a given date range when there is an existing block' do
+
+      hotel_booker.create_a_block(today + 2, today + 7, 5, 150)
+
+      expect(hotel_booker.list_available_rooms(Hotel::DateRange.new(today + 2, today + 7))).must_equal [*8..20]
+    end
+
+    it 'raises an error if given a check-out date occuring after a check-in date' do
+      expect{hotel_booker.list_available_rooms(today + 1, today)}.must_raise ArgumentError
+    end
+  end
+
+  describe 'list_unavailable_rooms helper method' do
+
+    it 'returns an array containing a list of unavailable room numbers for a given date range' do
+      expect(hotel_booker.list_available_rooms(Hotel::DateRange.new(today + 2, today + 4))).must_be_instance_of Array
+    end
+
+    it 'correctly identifies the list of unavailable room numbers for a given date range' do
+
+      room_1 = [1]
+      room_2 = [2]
+      both_rooms = room_1 + room_2
+
+      expect(hotel_booker.list_unavailable_rooms(captures_both_res_dates)).must_equal both_rooms
+      expect(hotel_booker.list_unavailable_rooms(overlap_res_1)).must_equal room_1
+      expect(hotel_booker.list_unavailable_rooms(overlap_res_2)).must_equal room_2
```

Comment:
In conjunction with a comment above:

This is one case in which I would've rather just had `overlap_res_2` defined locally here within this same test.